torch/distributed/distributed_c10d.py

```diff
@@ -753,6 +766,94 @@ def recv(tensor,
         return src
 
 
+class P2POp(object):
```

Comment:
Since we don't have test coverage for OSS on this, shall we mark it as a private API for now (start with an underscore) and make sure its doc does not show in our API page?